src/targets/gpu/gemm_impl.cpp

```diff
@@ -27,6 +28,22 @@ rocblas_datatype get_type(shape::type_t type)
     MIGRAPHX_THROW("ROCBLAS_GEMM: data type not supported!");
 }
 
+void blas_shape(const shape& s)
+{
+    if(s.lens().size() < 2)
```

Comment:
It is checking for size() < 2 and size() < 3, returning both times, Should it only check size() < 3 ?

Reply:
No it should return yet. If the size is at least two we need to check the memory layout still,